src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -1,488 +1,460 @@
+
+    val toDelete =
+        currentCountries.stream().filter(c -> !countryEntities.contains(c)).collect(toSet());
+    val toAdd =
+        countryEntities.stream()
+            .filter(c -> !currentCountries.contains(c))
+            .map(c -> createProgramCountry(programToUpdate, c))
+            .map(Optional::get)
+            .collect(toSet());
+
+    programToUpdate
+        .getProgramCountries()
+        .removeIf(programCountryPredicate(programToUpdate, toDelete));
+    currentCountries.forEach(
+        c -> c.getProgramCountries().removeIf(programCountryPredicate(programToUpdate, toDelete)));
+    programCountryRepository.saveAll(toAdd);
+  }
+
+  /**
+   * Compares the user provided list against the list stored in the system. Throws an exception if
+   * it determines that the users provided a collection that cannot be found by the system.
+   *
+   * <p>Precondition: System List should always be a subset of User List
+   *
+   * @param errorTemplate String template for the exception message.
+   * @param userList The user provided collection.
+   * @param systemList The system collection.
+   */
+  @SuppressWarnings("unchecked")
+  static void compareLists(String errorTemplate, List userList, List systemList) {
+    if (userList.size() > systemList.size()) { // Couldn't find all of them
+      val badInput = new StringBuilder();
+      userList.forEach(
+          item -> {
+            if (!systemList.contains(item)) {
+              badInput.append(" ");
+              badInput.append(item.toString());
+            }
+          });
+      val message = String.format(errorTemplate, badInput.toString());
+      throw Status.INVALID_ARGUMENT.augmentDescription(message).asRuntimeException();
+    }
+  }
+
+  public void removeProgram(String name) throws EmptyResultDataAccessException {
+    val p = getProgram(name);
+    programRepository.deleteById(p.getId());
+    log.info("Program {} is successfully deleted. ", name);
+  }
+
+  public List<ProgramEntity> listPrograms() {
+    val programs =
+        programRepository.findAll(
+            new ProgramSpecificationBuilder()
+                .setFetchCancers(true)
+                .setFetchPrimarySites(true)
+                .setFetchInstitutions(true)
+                .setFetchCountries(true)
+                .listAll(true));
+    return List.copyOf(programs);
+  }
+
+  public List<String> getAllProgramNames() {
+    val programNames = programRepository.getActivePrograms();
+    return programNames;
+  }
+
+  List<CancerEntity> listCancers() {
+    return List.copyOf(cancerRepository.findAll());
+  }
+
+  List<PrimarySiteEntity> listPrimarySites() {
+    return List.copyOf(primarySiteRepository.findAll());
+  }
+
+  List<CountryEntity> listCountries() {
+    return List.copyOf(countryRepository.findAll());
+  }
+
+  List<RegionEntity> listRegions() {
+    return List.copyOf(regionRepository.findAll());
+  }

```

Comment:
Still needed?

Reply:
listRegions() method is being used by getRegions RPC to fetch the regions